InvisibleMap/ViewController.swift

```diff
@@ -189,6 +401,41 @@ class ViewController: UIViewController {
         tagFinderTimer = Timer.scheduledTimer(timeInterval: 0.2, target: self, selector: #selector(self.updateLandmarks), userInfo: nil, repeats: true)
     }
     
+    func scheduledPingTimer() {
+        self.pingTimer.invalidate()
+        self.pingTimer = Timer()
+        self.pingTimer = Timer.scheduledTimer(timeInterval: 0.5, target: self, selector: #selector(self.ping), userInfo: nil, repeats: true)
+    }
+    
+    @objc func ping() {
+        if !self.firstTagFound {
+            return
+        }
+        self.playSound(type: "ping")
+        let volume = self.audioPlayers["ping"]!!.volume
+        let hapticScale = pow(volume, 1.0 / 3.0)
+        if hapticScale > 0.75 {
```

Comment:
add a comment explaining this relationship and why you chose it (I get it, but I think it's a little bit mysterious without the comment)